xla/service/while_loop_all_reduce_code_motion.cc

```diff
@@ -961,6 +963,19 @@ absl::StatusOr<bool> WhileLoopAllReduceCodeMotion::Run(
                         HloReplicationAnalysis::RunWithPartialReplication(
                             module, /*cross_partition_spmd=*/true));
   }
+
+  // Run setup passes that may setup the add(all-reduce/reduce-scatter,
```

Comment:
Can you elaborate what you mean with setup passes? What is being set up here?

Reply:
See comments above we are setting up the 

add(all-reduce/reduce-scatter, get-tuple-element(while_loop_param)) pattern